src/stores/obsPortalData.js

```diff
@@ -10,49 +10,86 @@ export const useObsPortalDataStore = defineStore('obsPortalData', {
     return {
       completedObservations: {},
       upcomingRealTimeSessions: {},
-      pendingRequestGroups: [],
+      pendingScheduledObservations: {},
       observationDetails: {},
       selectedConfiguration: null
     }
   },
   persist: true,
   actions: {
-    sortResponseData (response) {
-      const currentTime = new Date().toISOString()
-      for (const result of response.results) {
-        const sessionEnd = new Date(result.end).toISOString()
-        if ((result.state === 'COMPLETED') || (result.observation_type === 'REAL_TIME' && sessionEnd < currentTime)) {
-          if (!this.completedObservations[result.id]) {
-            this.completedObservations[result.id] = result
-          }
-        } else if (result.observation_type === 'REAL_TIME' && sessionEnd > currentTime) {
-          if (!this.upcomingRealTimeSessions[result.id]) {
-            this.upcomingRealTimeSessions[result.id] = result
-          }
+    storeUpcomingRealTimeSessions (realTimeSessions) {
+      for (const session of realTimeSessions.results) {
+        if (!this.upcomingRealTimeSessions[session.id]) {
+          this.upcomingRealTimeSessions[session.id] = session
         }
       }
     },
-    async fetchCompleteObservationsAndUpcomingRTSessions () {
+    async fetchUpcomingRealTimeSessions () {
       const configurationStore = useConfigurationStore()
       const userDataStore = useUserDataStore()
       const username = userDataStore.username
+      const now = new Date().toISOString()
       await fetchApiCall({
-        url: configurationStore.observationPortalUrl + `observations/?user=${username}&state=PENDING&state=COMPLETED&limit=100&ordering=-start`,
+        url: configurationStore.observationPortalUrl + `observations/?user=${username}&state=PENDING&observation_type=REAL_TIME&limit=100&ordering=-start&end_after=${now}`,
         method: 'GET',
         successCallback: (response) => {
-          this.sortResponseData(response)
+          this.storeUpcomingRealTimeSessions(response)
         }
       })
     },
-    fetchPendingRequestGroups () {
+    storePendingScheduledObservations (response) {
+      // The format of the response is as follows:
+      // [
+      //     {
+      //       "id": 680316844,
+      //       "request": {
+      //           "id": 3784722,
+      //           "state": "PENDING",
+      //           "configurations": [
+      //             {...configuration details...}
+      //           ]
+      //       },
+      //       and other fields
+      //   }
+      // ]
+      for (const pendingScheduledObservation of response.results) {
+        // Because a scheduled request is ephemeral and its id can change, we store the request id which is stable
+        // So the example above would be stored as:
+        // {
+        //   3784722: {... details of the request ...}
+        if (!this.pendingScheduledObservations[pendingScheduledObservation.request.id]) {
```

Comment:
This will not update the observation details (where and when it is scheduled) when that changes each scheduling run. Basically I think you need to just get rid of the if statement here, and always set the observation for this request id so you always have the latest scheduled observation details.